### PR TITLE
TFKUBE-446: Unify NFS usage across all products

### PR DESCRIPTION
**Pull request by nanux**
## Pull request description

* Switch Bitbucket to use `persistentVolumeClaim` instead of accessing NFS directly
* Remove `capacity` that was causing confusion
* Remove the conditional for the shared home PV and PVC creation
* Rename variables for consistency and clarity
* Fix unit tests

## Checklist
- [x] I have successful end to end tests run (with & without domain)
- [x] I have added unit tests (if applicable)
- [x] I have user documentation (if applicable)

### CODE REVIEW

**Comment by nanux** [comment]
E2E tests
https://github.com/atlassian-labs/data-center-terraform/actions/runs/2225288204
https://github.com/atlassian-labs/data-center-terraform/actions/runs/2225288965

File: modules/products/bitbucket/helm.tf [hunk 1]
```diff
@@ -50,17 +50,11 @@ resource "helm_release" "bitbucket" {
...
-              path   = "/srv/nfs"
+          customVolume = {
+            persistentVolumeClaim = {
+              claimName = module.nfs.nfs_claim_name
             }
           }
-          persistentVolumeClaim = {
-            create           = true
-            storageClassName = ""
-          }
```
Comment (errcode1202) [inline modules/products/bitbucket/helm.tf:63]: To be honest, this is now so much cleaner too 👍🏽

**Review by errcode1202 — APPROVED** [review]

File: modules/AWS/nfs/outputs.tf [hunk 2]
```diff
@@ -3,5 +3,5 @@ output "helm_release_nfs_service_ip" {
 }
 
 output "nfs_claim_name" {
-  value = var.shared_home_size == null ? null : kubernetes_persistent_volume_claim.shared-home-pvc[0].metadata[0].name
+  value = var.shared_home_size == null ? null : kubernetes_persistent_volume_claim.product_shared_home_pvc.metadata[0].name
```
Comment (jjeongatl) [inline modules/AWS/nfs/outputs.tf:6]: `var.shared_home_size == null ?`
Do we still need this condition?

File: modules/AWS/nfs/variables.tf [hunk 3]
```diff
@@ -58,7 +52,7 @@ variable "shared_home_snapshot_id" {
     error_message = "Provide correct EBS snapshot ID."
   }
 }
-# The size of the EBS volume to allocate for shared home. If is null no PV and PVC will be created.
+
 variable "shared_home_size" {
```
Comment (nghazali) [inline modules/AWS/nfs/variables.tf:56]: Please remove default value as this is not valid anymore. Also add validation for shared_home please.

**Review by nghazali — CHANGES_REQUESTED** [review]

File: modules/AWS/nfs/outputs.tf (hunk 2 above)
Comment (nanux) [reply modules/AWS/nfs/outputs.tf:6]: We don't, good catch!

File: modules/AWS/nfs/variables.tf [hunk 4]
```diff
@@ -58,9 +52,12 @@ variable "shared_home_snapshot_id" {
...
-  description = "The storage capacity to allocate to local home"
+  description = "The storage capacity to allocate to shared home"
   type        = string
-  default     = null
+  validation {
+    condition     = can(regex("^[0-9]+([gG]|Gi)$", var.shared_home_size)) || var.shared_home_size == null
```
Comment (nanux) [inline modules/AWS/nfs/variables.tf:60]: We support only G units due to this line - https://github.com/atlassian-labs/data-center-terraform/pull/217/files#diff-26d8f7b099b5c1c21aeafca0e13eb3676ace8a8bd69e0d240a7446b8cbe8c2efR5

**Review by jjeongatl — APPROVED** [review]

File: modules/AWS/nfs/variables.tf (hunk 4 above)
Comment (nghazali) [reply modules/AWS/nfs/variables.tf:60]: Null value for `shared_home` should not be valid

**Review by nghazali — CHANGES_REQUESTED** [review]

**Review by nghazali — APPROVED** [review]

File: modules/common/variables.tf [hunk 5]
```diff
@@ -47,6 +47,15 @@ variable "namespace" {
   }
 }
 
+variable "shared_home_size" {
+  description = "Shared home persistent volume size."
+  type        = string
+  validation {
+    condition     = can(regex("^[0-9]+([gG]|Gi)$", var.shared_home_size)) || var.shared_home_size == null
```
Comment (nghazali) [inline modules/common/variables.tf:54]: Do still we need this `null` in validation.

**Pull request merged**